src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -0,0 +1,95 @@
+---
+layout: article
+title: Develop locally
+description: Learn to develop Appwrite functions locally.
+---
+
+Develop your Appwrite functions locally to make code changes without redeploying your function on every code change and hot reload your code for faster testing.
+
+# Setup {% #setup %}
+
+1. Install the [Docker CLI](https://www.docker.com/products/docker-desktop/)
+2. Ensure Docker is running in the background
+3. Install the [Appwrite CLI](/docs/tooling/command-line/installation#installation)
+4. [Log in](/docs/tooling/command-line/installation#get-started) to your Appwrite account using `appwrite login`
+5. [Initialize your project](/docs/tooling/command-line/installation#initialization) and create the `appwrite.json` file
+
+We use Docker to replicate the production environment for the local deployment of functions. These can be executed locally with the `appwrite run functions` CLI command, which requires initializing a project with an `appwrite.json` file. The CLI also supports various other [CLI commands](/docs/tooling/command-line/commands).
+
+{% arrow_link href="/docs/tooling/command-line/functions" %}
+Learn more about deploying functions using CLI
+{% /arrow_link %}
```

Comment:
> The CLI also supports various other [CLI commands](/docs/tooling/command-line/commands).
>
> {% arrow_link href="/docs/tooling/command-line/functions" %}
> Learn more about deploying functions using CLI
> {% /arrow_link %}

Let's not take them away from this page.

> We use Docker to replicate the production environment for the local deployment of functions. These can be executed locally with the `appwrite run functions` CLI command, which requires initializing a project with an `appwrite.json` file.

It sounds like this paragraph does a little bit of explanation of the numbered list. As such, let's move it before the numbered list to explain first and then give the directions. Also, for the explanation, let's not mention the `appwrite run functions` to avoid mentioning something without explanation.

Lastly, it might be good to clarify there must be code locally in order to run `appwrite run function`.